### DISCUSSION TITLE
Decrease chances that mail is considered spam

### DISCUSSION THREAD

**Discussion by be-joeri** [mealie-recipes/mealie#2993] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

First of all: thumbs up for this project! Nice piece of software, up and running in no time.

I had an issue with mails sent by Mealie. My mail server rejected them as spam and I had to tweak my spam filter (rspamd) to accept the mails.

These are some of the most important reasons the filter marked the mail as spam:
- MISSING_MID: the mail does not have a Message-ID header
- MISSING_MIME_VERSION: MIME-Version header is missing
- MIME_MA_MISSING_TEXT: MIME multipart/alternative missing text/plain part
- MISSING_DATE: Date header missing

Fixing this can improve deliverability for people with strict spam filters.

### Please provide a concise description of the feature that would resolve your issue.

Adding a Message-ID and Date header would already improve the chances of the mail being accepted.
Considering adding a plain text version next to the html version would even improve this more.

I'm not an expert, but I think smtplib offers the tools to do this.

### Please consider and list out some caveats or tradeoffs made in your design decision

-

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
Date will be included when #3029 is merged
